ibis/expr/types/relations.py

```diff
@@ -738,7 +738,13 @@ def filter(
         child = self
         for predicate, right in top_ks:
             child = child.semi_join(right, predicate)[child]
-        return an.apply_filter(child, resolved_predicates)
+        return an.apply_filter(
```

Comment:
What exactly happens here? :)

Reply:
Ultimately a `Selection` is constructed. Is there something specific you're wanting to know about?

Reply:
The point of this rewrite is to turn expressions like `a > a.mean()` into an expression that is meaningful to SQL.